### PR TITLE
FINERACT-1706: Loan summary with principal adjustments

### PR DESCRIPTION
**Pull request by josehernandezfintecheandomx**
## Description

As a user when I do a chargeback transaction, the repayment schedule to be recalculated from the chargeback date and amount and the loan summary must to be updated in order to reflect the new chargeback amount

[FINERACT-1706](https://issues.apache.org/jira/browse/FINERACT-1706).

Screen Shot 2022-10-14 at 14 50 45

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Review by jdailey — COMMENTED** [review]
The check boxes are not marked and I do not see test coverage for this feature.  That includes, how is this change to be verified as backward compatible with existing features.  

There are some lines that seem to be unchanged but show up as - and then +, thus making the change hard to track.

File: fineract-provider/src/main/resources/db/changelog/tenant/parts/0056_add_principal_adjustments_to_loan.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    KIND, either express or implied. See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.1.xsd">
+    <changeSet author="fineract" id="1">
+        <addColumn tableName="m_loan">
+            <column name="principal_adjustments_derived" type="DECIMAL(19, 6)">
+                <constraints nullable="false"/>
```
Comment (galovics) [inline fineract-provider/src/main/resources/db/changelog/tenant/parts/0056_add_principal_adjustments_to_loan.xml]: If this is a not nullable column, either specify a default value or just make it nullable.

**Review by galovics — CHANGES_REQUESTED** [review]
No tests.

File: fineract-provider/src/main/resources/db/changelog/tenant/parts/0056_add_principal_adjustments_to_loan.xml (hunk 1 above)
Comment (josehernandezfintecheandomx) [reply fineract-provider/src/main/resources/db/changelog/tenant/parts/0056_add_principal_adjustments_to_loan.xml]: Done! added defaultValueNumeric="0.000000"

**Review by galovics — APPROVED** [review]

**Pull request merged**